utility/session_test.go

```diff
@@ -0,0 +1,531 @@
+		require.Equal(t, len(session1.Fishermen), len(session2.Fishermen))
+		require.Equal(t, len(session1.Fishermen), len(session3.Fishermen))
+
+		// Assert different services between apps
+		assertActorsDifference(t, session1.Servicers, session2.Servicers, probabilityOfOverlap)
+		assertActorsDifference(t, session1.Servicers, session3.Servicers, probabilityOfOverlap)
+
+		// Assert different fishermen between apps
+		assertActorsDifference(t, session1.Fishermen, session2.Fishermen, probabilityOfOverlap)
+		assertActorsDifference(t, session1.Fishermen, session3.Fishermen, probabilityOfOverlap)
+
+		if numBlocksUntilChange == 0 {
+			// Assert different servicers between heights for the same app
+			assertActorsDifference(t, app1PrevServicers, session1.Servicers, probabilityOfOverlap)
+			assertActorsDifference(t, app2PrevServicers, session2.Servicers, probabilityOfOverlap)
+			assertActorsDifference(t, app3PrevServicers, session3.Servicers, probabilityOfOverlap)
+
+			// Assert different fishermen between heights for the same app
+			assertActorsDifference(t, app1PrevFishermen, session1.Fishermen, probabilityOfOverlap)
+			assertActorsDifference(t, app2PrevFishermen, session2.Fishermen, probabilityOfOverlap)
+			assertActorsDifference(t, app3PrevFishermen, session3.Fishermen, probabilityOfOverlap)
+
+			// Store the new servicers and fishermen for the next height
+			app1PrevServicers = session1.Servicers
+			app2PrevServicers = session2.Servicers
+			app3PrevServicers = session3.Servicers
+			app1PrevFishermen = session1.Fishermen
+			app2PrevFishermen = session2.Fishermen
+			app3PrevFishermen = session3.Fishermen
+
+			// Reset the number of blocks until we expect a different set of servicers and fishermen
+			numBlocksUntilChange = numBlocksPerSession - 1
+		} else {
+			// Assert the same servicers between heights for the same app
+			require.ElementsMatch(t, app1PrevServicers, session1.Servicers)
+			require.ElementsMatch(t, app2PrevServicers, session2.Servicers)
+			require.ElementsMatch(t, app3PrevServicers, session3.Servicers)
+
+			// Assert the same fishermen between heights for the same app
+			require.ElementsMatch(t, app1PrevFishermen, session1.Fishermen)
+			require.ElementsMatch(t, app2PrevFishermen, session2.Fishermen)
+			require.ElementsMatch(t, app3PrevFishermen, session3.Fishermen)
+
+			numBlocksUntilChange--
+		}
+
+		// Advance block height
+		writeCtx, err := persistenceMod.NewRWContext(height)
+		require.NoError(t, err)
+		err = writeCtx.Commit([]byte(fmt.Sprintf("proposer_height_%d", height)), []byte(fmt.Sprintf("quorum_cert_height_%d", height)))
+		require.NoError(t, err)
+		writeCtx.Release()
+	}
+}
+
+func TestSession_GetSession_ApplicationUnbonds(t *testing.T) {
+	// TODO: What if an Application unbonds (unstaking period elapses) mid session?
+}
+
+func TestSession_GetSession_ServicersAndFishermenCounts_GeoZoneAvailability(t *testing.T) {
+	// TODO: Once GeoZones are implemented, the tests need to be added as well
```

Comment:
Is there a more relevant issue?

```suggestion
	// TODO(#697): Once GeoZones are implemented, the tests need to be added as well
```